inference/core/utils/sqlite_wrapper.py

```diff
@@ -0,0 +1,266 @@
+import os
+import sqlite3
+from typing import Any, Dict, List, Optional
+
+from inference.core.logger import logger
+
+ColName = str
+ColType = str
+ColValue = str
+
+
+class SQLiteWrapper:
+    def __init__(
+        self,
+        db_file_path: str,
+        table_name: str,
+        columns: Dict[ColName, ColType],
+        connection: Optional[sqlite3.Connection] = None,
+    ):
+        self._db_file_path = db_file_path
+        self._tbl_name = table_name
+        self._columns = {**columns, **{"id": "INTEGER PRIMARY KEY"}}
+
+        if not connection:
+            if not os.path.exists(os.path.dirname(db_file_path)):
+                os.makedirs(os.path.dirname(db_file_path))
+            connection: sqlite3.Connection = sqlite3.connect(db_file_path, timeout=1)
+            self.create_table(connection=connection)
+            connection.close()
+        else:
+            self.create_table(connection=connection)
+
+    def create_table(self, connection: Optional[sqlite3.Connection] = None):
+        if not connection:
+            connection: sqlite3.Connection = sqlite3.connect(
+                self._db_file_path, timeout=1
+            )
+            self._create_table(connection=connection)
+            connection.close()
+        else:
+            self._create_table(connection=connection)
+
+    def _create_table(self, connection: sqlite3.Connection):
+        sql_create_table = f"""CREATE TABLE IF NOT EXISTS {self._tbl_name}
+                ({', '.join(f'{_col} {_type}' for _col, _type in self._columns.items())});
+            """
+        cursor = connection.cursor()
+        cursor.execute("BEGIN EXCLUSIVE")
+        try:
+            cursor.execute(sql_create_table)
+            connection.commit()
+            cursor.close()
+        except Exception as exc:
+            connection.rollback()
+            cursor.close()
+            raise exc
+
+    def insert(
+        self,
+        values: Dict[ColName, ColValue],
+        connection: Optional[sqlite3.Connection] = None,
+    ):
```

Comment:
could u give return types annotations?

Reply:
Done, this method should not return, modified a bit to re-rise instead.